src/transaction.cairo

```diff
@@ -135,6 +202,31 @@ pub impl TransactionImpl of TransactionTrait {
     fn serialize_no_witness(self: Transaction) -> ByteArray {
         self.btc_encode(BASE_ENCODING)
     }
+
+    fn calculate_block_subsidy(block_height: u32) -> i64 {
+        let halvings = block_height / 210000;
+        if halvings >= 64 {
+            return 0;
+        }
+        utils::shr::<i64, u32>(50 * 100000000, halvings)
+    }
+
+    fn validate_coinbase(tx: Transaction) -> bool {
```

Comment:
Can you make this function return a `Result<(),felt252>`, and add some errors to `src/errors.cairo` for invalid coinbase?